File: front/service/item/Item.service.js
import axios from "axios";
import { config } from "../../src/config/config";
import authHeader from '../user/auth-header'

const API_URL = config+"api/item/";

class ItemService {
    createItemUser(id, data) {
         return axios.post(API_URL + "createItemUser/"+id, data,  { headers: { Authorization:  authHeader() }});
    } 
    getItemAll() {
        return axios.get(API_URL + "non/getItemAll", null, { headers: { Authorization:  authHeader() }});
    } 
    getItemById(itemId) {
        return axios.get(API_URL + "getItemById/"+itemId,  { headers: { Authorization:  authHeader() }});
    } 
    updateItemById(id, data) {
        return axios.patch(API_URL + "updateItemById/"+id, data,  { headers: { Authorization:  authHeader() }});
   } 
    deleteItemById(itemId) {
        return axios.delete(API_URL + "deleteItemById/"+itemId, { headers: { Authorization:  authHeader() }});
    }  
    likeItemAndUserById(itemId,userId) {
        return axios.post(API_URL + "likeItemAndUserById/"+itemId+"/"+userId, null,  { headers: { Authorization:  authHeader() }});
    } 
    unlikeItemAndUserById(itemId,userId) {
        return axios.delete(API_URL + "unlikeItemAndUserById/"+itemId+"/"+userId, { headers: { Authorization:  authHeader() }});
    }
}

export default new ItemService();

File: back/src/main/java/call/web/demo/service/ItemService.java
package call.web.demo.service;

import call.web.demo.model.Item;
import call.web.demo.repository.item.dto.ItemRequestDto;
import call.web.demo.repository.item.dto.ItemResponseDto;
import call.web.demo.repository.user.dto.MessageResponse;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface ItemService {

    List<ItemResponseDto> getItemAll();
    ResponseEntity<?> getItemById(Long itemId);
    ResponseEntity<MessageResponse> updateItemById(Long itemId, ItemRequestDto itemDto);
    void createItemUser(Long user_id, ItemRequestDto itemDto);
    ResponseEntity<?> deleteItemById(Long itemId);
    ResponseEntity<?> likeItemAndUserById(Long itemId, Long userId);
    ResponseEntity<?> unlikeItemAndUserById(Long itemId, Long userId);
}
